substrate/client/chain-spec/src/chain_spec.rs

```diff
@@ -448,23 +451,29 @@ impl<G, E> ChainSpecBuilder<G, E> {
 		ChainSpec {
 			client_spec,
 			genesis: GenesisSource::GenesisBuilderApi(self.genesis_build_action, self.code.into()),
+			_host_functions: Default::default(),
```

Comment:
The build should return a `ChainSpec` instance the proper HF.
Maybe something like:

```
fn build<HF: HostFunctions>(self) -> ChainSpec<G, E, HF> {
    ...
    ChainSpec {
       ...
       _host_functions: PhantomData,
    }
}
```

Or (maybe better) the bbuilder should be generic over HF as well and the we have something like:
```
impl<G,E,HF> ChainSpec<G,E,HF> {
  ...
  fn builder(..) -> ChainSpecBuilder<G, E, HF> {
     ...
  }
```

Reply:
done in https://github.com/paritytech/polkadot-sdk/pull/2190/commits/c405f5a7fed346f544fb1c0c67f8e1a37a5dd89f

Reply:
You need to set the `build()` return type to `ChainSpec<G, E, EHF>`